### PR TITLE
Fix loaded values in permission dialog for participants

### PR DESCRIPTION
**Pull request by danxuliu**
The permissions dialog for participants was using `attendeePermissions` rather than `permissions`, so the dialog showed the default permissions for participants without specific permissions even if conversation permissions were set.

Besides that, the participant list is now reloaded on conversation and call permissions change, as due to the permission chain the actual participant permissions may have changed.

Finally, I have added some integration tests for setting conversation permissions when participant permissions are set; I found a bit surprising that this resets the participant permissions, so better have an explicit test for that behaviour :-)

## How to test (scenario 1)

- Join a conversation as a moderator
- If there is none, add a regular user to the conversation
- Open the conversation settings
- Set some custom permissions in the conversation
- Close the conversation settings
- Reload the page
- In the sidebar, open the permission settings for the regular user

### Result with this pull request

The permissions shown for the participant are the conversation permissions

### Result without this pull request

The permissions shown for the participant are the original default permissions, which do not match the actual permissions of the user (which are the conversation permissions)

## How to test (scenario 2)

- Join a conversation as a moderator
- If there is none, add a regular user to the conversation
- Set some custom permissions for the regular user
- Open the conversation settings
- Set some custom permissions in the conversation

### Result with this pull request

The icon about custom permissions for the user is removed

### Result without this pull request

The icon about custom permissions for the user is kept, even if setting the conversation permissions reset the user permissions

### CODE REVIEW

File: src/components/RightSidebar/Participants/ParticipantsList/Participant/ParticipantPermissionsEditor/ParticipantPermissionsEditor.vue [hunk 1]
```diff
@@ -86,11 +86,11 @@ export default {
 		},
 
 		/**
-		 * Permisssions of the current participant, from the participants
-		 * store.
+		 * Combined final permisssions of the current participant, from the
```
Comment (nickvergessen) [inline src/components/RightSidebar/Participants/ParticipantsList/Participant/ParticipantPermissionsEditor/ParticipantPermissionsEditor.vue:89]:
```suggestion
		 * Combined final permissions of the current participant, from the
```

File: tests/integration/features/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -584,7 +590,7 @@ private function mapPermissionsTestInput($permissions): int {
 	private function mapPermissionsAPIOutput($permissions): string {
 		$permissions = (int) $permissions;
 
-		$permissionsString = '';
+		$permissionsString = !$permissions? 'D': '';
```
Comment (nickvergessen) [inline tests/integration/features/bootstrap/FeatureContext.php:593]:
```suggestion
		$permissionsString = !$permissions ? 'D' : '';
```

**Review by nickvergessen — APPROVED** [review]
Time to fixup

**Comment by danxuliu** [comment]
Integration test failures are unrelated to this pull request and caused by https://github.com/nextcloud/server/commit/22875bf367b69db4a76daed1a057407e7fbff2a6

**Pull request merged**